stablehlo/reference/Ops.cpp

```diff
@@ -233,6 +233,57 @@ Tensor evalFloorOp(const Tensor &operand, TensorType resultType) {
   return result;
 }
 
+Tensor evalGatherOp(const Tensor &operand, const Tensor &startIndices,
```

Comment:
The current implementation does not seem to match the spec. I can roughly understand what the current implementation is doing, but it may not be obvious to users familiar with the spec and expecting strong correlation between the two artifacts. Here are some of the non-obvious differences:
1.   `nonCollapsedSliceDims` to store the axes of operand which are not mentioned in the `collapsedSliceDims`. 
2. Use of clamp (which IMO we should not need if follow the spec).
3. Starting with startIndices to derive the input/output indices.   

IMO, implementing the spec, as it is, should be possible in this case. One non-trivial aspect could be 
```
start_index = start_indices[bi0, ..., :, ..., biN]
``` 

but can be implemented using a slice op with start index as the  `batch_indices` and limit index should be `[bi0, ..., , bik + dim(start_indices, index_vector_dims), ... biN] `, wherte  `k =  index_vector_dims`

Reply:
Thanks for the suggestion updated PR.